synapse/handlers/message.py

```diff
@@ -432,6 +432,9 @@ def __init__(self, hs: "HomeServer"):
 
         self._ephemeral_events_enabled = hs.config.enable_ephemeral_messages
 
+        self._external_cache = hs.get_external_cache()
+        self._use_external_cache = hs.config.redis.redis_enabled
```

Comment:
I wonder if this should be `redis enabled and more than one federation sender`?

Reply:
Or at the very least if the federation sending is done by the main process it probably doesn't make sense to use this?

Reply:
Maybe, though it is worth noting that it's a) probably cheaper to calculate it on event creator, as you'll already have the state in the cache, and b) means that you can run multiple event creators to split up the work without having to run multiple federation senders.

Reply:
Ah, right that makes sense. I guess it is a reasonable assumption that anyone running workers would be running at least one of an event creator or federation sender?

Reply:
Yeah, and its really not the end of the world if they aren't.